content/6_Advanced/String_Search.mdx

```diff
@@ -99,7 +99,31 @@ vector<int> pi(const string &s) {
 ```
 
 </CPPSection>
+<PySection>
+
+```py
+def solve(s):
+	n = len(s)
+	pi = [0]*n
+	j = 0
+	for i in range(1,n):
+		while(j > 0 and s[j]!=s[i]):
+			j = pi[j-1]
+		if s[i] == s[j]:
+			j+=1
+		pi[i] = j
+	return pi
+	
+def KMP(s,t):
```

Comment:
whats this function about?

Reply:
This Function basically acts as a main function of the question. It takes 2 strings as an input and returns integer as a string.
(Written in the comments section)